file/lib/src/sentry_io_overrides_integration.dart

```diff
@@ -0,0 +1,32 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:sentry/sentry.dart';
+import 'sentry_io_overrides.dart';
+
+/// When installed, every new file will be created as [SentryFile].
+/// When installed, operations will use [SentryFile] instead of dart:io's [File]
+/// implementation whenever [File] is used.
+///
+/// Must be called outside the RunZoneGuardedIntegration.
+/// Otherwise the FileOverrides aren't attached to the root zone.
+class SentryIOOverridesIntegration extends Integration<SentryOptions> {
+  IOOverrides? _previousOverrides;
+  SentryOptions? _options;
+  @override
+  FutureOr<void> call(Hub hub, SentryOptions options) {
+    _options = options;
+    if (options.isTracingEnabled()) {
+      _previousOverrides = IOOverrides.current;
+      IOOverrides.global = SentryIOOverrides(hub);
+      options.sdk.addIntegration('sentryIOOverridesIntegration');
+    }
+  }
+
+  @override
+  FutureOr<void> close() {
+    if (_options?.isTracingEnabled() ?? false) {
```

Comment:
If the SDK is closed, depending on the order of closing the integrations, this will always return false, maybe having a boolean `_installed` would be better here.